contracts/chain-adapters/l2/WithdrawalAdapterBase.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { MultiCaller } from "@uma/core/contracts/common/implementation/MultiCaller.sol";
+import { CircleCCTPAdapter, ITokenMessenger, CircleDomainIds } from "../../libraries/CircleCCTPAdapter.sol";
+
+/**
+ * @title WithdrawalAdapterBase
+ * @notice This contract contains general configurations for bridging tokens from an L2 to a single recipient on L1.
+ * @dev This contract should be deployed on L2. It provides an interface to withdraw tokens to some address on L1. The only
+ * function which must be implemented in contracts which inherit this contract is `withdrawToken`. It is up to that function
+ * to determine which bridges to use for an input L2 token. Importantly, that function must also verify that the l2 to l1
+ * token mapping is correct so that the bridge call itself can succeed.
+ */
+abstract contract WithdrawalAdapterBase is CircleCCTPAdapter, MultiCaller {
+    using SafeERC20 for IERC20;
```

Comment:
Prefer naming this contract WithdrawHelper or L3WithdrawHelper as we've typically used Adapter to contain code that gets delegate called and otherwise acts as a library

Reply:
https://github.com/across-protocol/contracts/pull/605/commits/da1bc9abb9f9f75e83652b7e9b5be7ba41e1dca3